hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -455,7 +459,26 @@ protected DataMapFilter getFilterPredicates(Configuration configuration) {
     try {
       String filterExprString = configuration.get(FILTER_PREDICATE);
       if (filterExprString == null) {
-        return null;
+        String expr = configuration.get(TableScanDesc.FILTER_EXPR_CONF_STR);
+        if (expr == null) {
+          return null;
+        }
+        ExprNodeGenericFuncDesc exprNodeGenericFuncDesc =
+            Utilities.deserializeObject(expr, ExprNodeGenericFuncDesc.class);
+        LOG.debug("hive expression:" + exprNodeGenericFuncDesc.getGenericUDF());
+        LOG.debug("hive expression string:" + exprNodeGenericFuncDesc.getExprString());
+        Expression expression =
+            Hive2CarbonExpression.convertExprHive2Carbon(exprNodeGenericFuncDesc);
```

Comment:
Can we do this filter expression conversion in MapredCarbonInputFormat.java (in carbondata-hive module) and set the FILTER_PREDICATE in the hadoop configuration, so that we donot need to add hive-exec dependency in carbondata-hadoop module.

Reply:
ok